pilot/pkg/proxy/envoy/watcher.go

```diff
@@ -293,41 +292,14 @@ func (proxy envoy) Run(config interface{}, epoch int, abort <-chan error) error
 	cmd := exec.Command(proxy.config.BinaryPath, args...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-
-	// Set if the caller is monitoring envoy, for example in tests or if envoy runs in same
-	// container with the app.
-	if proxy.errChan != nil {
-		// Caller passed a channel, will wait itself for termination
-		go func() {
-			proxy.errChan <- cmd.Wait()
-		}()
-		return nil
-	}
```

Comment:
@mandarjog I don't have an equivalent for this block of code yet.  AFAIK no test cases exercise it so removing it doesn't cause an issue, but I'm going to try to write some test cases for it against the unchanged version and see if I can get it covered, then make sure the new code passes the test cases.